sensu/plugins/metric-os-api.py

```diff
@@ -0,0 +1,150 @@
+#
+# return CRITICAL if any commands fail
+# return WARNING if no commands fail, but one or more timeout
+#
+#
+# Jose L Coello Enriquez <jlcoello@us.ibm.com>
+
+from argparse import ArgumentParser
+import re
+import sys
+import time
+import socket
+import os
+import json
+import requests
+
+STATE_OK = 0
+STATE_WARNING = 1
+STATE_CRITICAL = 2
+
+state = STATE_OK
+
+metric_data = ''
+
+DEFAULT_SCHEME = '{}'.format(socket.gethostname())
+
+
+def request_token(url, method='GET', retries=2, **kwargs):
+    r = None
+    try:
+        for i in range(retries + 1):
+            if i > 0:
+                time.sleep(2)
+            r = requests.request(method, url, **kwargs)
+            if r.status_code < 500:
+                break
+    except requests.exceptions.RequestException as e:
+        print "{0} returned {1}".format(url, e)
+        state = STATE_WARNING
+
+    return r
+
+def request(scheme, url, metric, timestamp, method='GET', retries=2, **kwargs):
+    global state
+    global metric_data
+
+    start = time.time()
+    r = None
+    try:
+        for i in range(retries + 1):
+            if i > 0:
+                time.sleep(2)
+            r = requests.request(method, url, **kwargs)
+            if r.status_code:
+                break
+    except requests.exceptions.RequestException as e:
+        print "{0} returned {1}".format(url, e)
+        state = STATE_WARNING
+
+    if r is None:
+        status = 0
+        status_code = 600
+        time_taken = -10
+        if state == STATE_OK:
+            state = STATE_WARNING
+    elif r.status_code != 200:
+        status = 0
+        status_code = r.status_code
+        time_taken = -5
+        state = STATE_CRITICAL
+    else:
+        status = 100
+        status_code = r.status_code
+        time_taken = round(time.time() - start, 3)
+
+    metric_data += '{0}.time.{1} {2} {3}\n'.format(scheme, metric, time_taken,
+                   timestamp)
+    metric_data += '{0}.code.{1} {2} {3}\n'.format(scheme, metric, status_code,
```

Comment:
This is sample output of a run : 
sla.monitor-controller1.time.glance-images 0.28 1440623138
sla.monitor-controller1.code.glance-images 200 1440623138
sla.monitor-controller1.status.glance-images 100 1440623138

I'm a bit confuse on what you are referring to?